bpf/aya-bpf/src/maps/array.rs

```diff
@@ -47,14 +47,27 @@ impl<T> Array<T> {
         }
     }
 
+    #[inline(always)]
     pub fn get(&self, index: u32) -> Option<&T> {
-        unsafe {
-            let value = bpf_map_lookup_elem(
-                self.def.get() as *mut _,
-                &index as *const _ as *const c_void,
-            );
-            // FIXME: alignment
-            NonNull::new(value as *mut T).map(|p| p.as_ref())
-        }
+        unsafe { self.lookup(index).map(|p| p.as_ref()) }
+    }
+
+    #[inline(always)]
+    pub fn get_ptr(&self, index: u32) -> Option<*const T> {
+        unsafe { self.lookup(index).map(|p| p.as_ptr() as *const T) }
```

Comment:
I think the extra cast isn't necessary here?

Reply:
It is necessary. To my surprise, `NonNull::as_ptr` returns... `*mut T`

https://doc.rust-lang.org/core/ptr/struct.NonNull.html#method.as_ptr